compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -18161,6 +18178,13 @@ private STNode getTypeDescFromExpr(STNode expression) {
                 return expression;
             case UNARY_EXPRESSION:
                 return STNodeFactory.createSingletonTypeDescriptorNode(expression);
+            case TUPLE_TYPE_DESC:
+                STTupleTypeDescriptorNode tupleTypeDescriptorNode = (STTupleTypeDescriptorNode) expression;
+                STNode memberNodes =
+                        STNodeFactory.createNodeList(getTypeDescList((STNodeList) tupleTypeDescriptorNode
+                                .memberTypeDesc));
+                return STNodeFactory.createTupleTypeDescriptorNode(tupleTypeDescriptorNode.openBracketToken,
+                        memberNodes, tupleTypeDescriptorNode.closeBracketToken);
```

Comment:
What puzzles me is, why do we create a tuple-type-desc with members being expressions in the first place?

We have `getTypeDescFromExpr()` but we seem to not have `getExprFromType()`. So once tuple-type-desc node is created it is a type. 

I think we need to call `expr = getTypeDescFromExpr(expr);` before creating a TupleTypeDescriptorNode at `parseTupleTypeDescOrExprStartsWithOpenBracket()`.

In fact, I tried running unit tests with the above change and they all are parsing. @rdulmina wdyt?

Reply:
+1